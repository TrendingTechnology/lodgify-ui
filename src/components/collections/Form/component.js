import React, { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { Card, Form } from 'semantic-ui-react';

import { Heading } from 'typography/Heading';
import { Button } from 'elements/Button';

/**
 * A form displays a set of related user input fields in a structured way.
 * @extends {React.PureComponent}
 */
export class Component extends PureComponent {
  state = {};

  persistInputChange = (name, value) => {
    this.setState({ [name]: value });
  };

  handleSubmit = () => {
    this.props.onSubmit(this.state);
  };

  cloneFormGroup = formGroup =>
    React.cloneElement(formGroup, {
      children: Children.map(formGroup.props.children, this.cloneInput),
      widths: 'equal',
    });

  cloneInput = input =>
    React.createElement(Form.Field, {
      children: React.cloneElement(input, {
        onChange: (name, value) => {
          this.persistInputChange(name, value);
          input.props.onChange(name, value);
        },
      }),
    });

  render = () => {
    const { children, headingText, submitButtonText, actionLink } = this.props;
    return (
      <Card>
        {headingText && (
          <Card.Content>
            <Heading size="small">{headingText}</Heading>
          </Card.Content>
        )}
        <Card.Content>
          <Form onSubmit={this.handleSubmit}>
            {Children.map(
              children,
              child =>
                child.type === Form.Group
                  ? this.cloneFormGroup(child)
                  : this.cloneInput(child)
            )}
            {actionLink && (
              <a onClick={actionLink.onClick}>{actionLink.text}</a>
            )}
            {submitButtonText && (
              <Button isPositionedRight>{submitButtonText}</Button>
            )}
          </Form>
        </Card.Content>
      </Card>
    );
  };
}

Component.displayName = 'Form';

Component.defaultProps = {
  headingText: null,
  onSubmit: Function.prototype,
  actionLink: null,
  submitButtonText: null,
};

Component.propTypes = {
  /** An optional action link. */
  actionLink: PropTypes.shape({
    /** The function to call when the secondary call to action is clicked. */
    onClick: PropTypes.func.isRequired,
    /** The visible text for the secondary call to action */
    text: PropTypes.string.isRequired,
  }),
  /** The child components and elements. */
  children: PropTypes.node.isRequired,
  /** The text to display as a heading at the top of the form. */
  headingText: PropTypes.string,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The text to display on the submit button. */
  submitButtonText: PropTypes.string,
};
