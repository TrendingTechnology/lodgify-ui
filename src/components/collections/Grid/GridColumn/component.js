import React from 'react';
import { Grid } from 'semantic-ui-react';

/**
 * GridColumn is the Lodgify UI interface for the
 * Semantic UI Grid.Column.
 *
 * @returns {Object}
 */
export const Component = props => <Grid.Column {...props} />;

Component.displayName = 'GridColumn';
