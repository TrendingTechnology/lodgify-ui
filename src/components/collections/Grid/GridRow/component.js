import React from 'react';
import { Grid } from 'semantic-ui-react';

import { DEFAULT_MAX_WIDTH } from '../constants';
import { getTwelveColumnsWidthValue } from '../utils';
/**
 * GridRow is the Lodgify UI 12-columns based component for the
 * Semantic UI Grid.Row.
 *
 * The [Semantic UI props](https://react.semantic-ui.com/collections/grid) can be for this component.
 * @returns {Object}
 */
export const Component = props => (
  <Grid.Row
    {...Object.assign({}, ...props, {
      columns: getTwelveColumnsWidthValue(DEFAULT_MAX_WIDTH),
    })}
  />
);

Component.displayName = 'GridRow';
