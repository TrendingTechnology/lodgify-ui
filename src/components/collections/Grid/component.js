import React from 'react';
import { Grid } from 'semantic-ui-react';

import { DEFAULT_MAX_WIDTH } from './constants';
import { getTwelveColumnsWidthValue } from './utils';
/**
 * Grid is the Lodgify UI 12-columns based component for the
 * Semantic UI Grid.
 * @type {Object}
 */
export const Component = props => (
  <Grid
    {...Object.assign({}, props, {
      columns: getTwelveColumnsWidthValue(DEFAULT_MAX_WIDTH),
    })}
  />
);

Component.displayName = 'Grid';
