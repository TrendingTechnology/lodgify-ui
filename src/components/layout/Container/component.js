import React from 'react';
import PropTypes from 'prop-types';
import { Container as SemanticContainer } from 'semantic-ui-react';

/**
 * Container is the Lodgify UI interface for the
 * Semantic UI Container.
 *
 * @returns {Object}
 */
export const Component = ({ textAlign, ...props }) => (
  <SemanticContainer {...props} textAlign={textAlign} />
);

Component.displayName = 'Container';

Component.defaultProps = {
  textAlign: 'left',
};

Component.propTypes = {
  /** The text align for nested children. */
  textAlign: PropTypes.oneOf(['left', 'center', 'right', 'justified']),
};
