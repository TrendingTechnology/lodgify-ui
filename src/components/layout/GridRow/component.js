import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

/**
 * GridRow is the Lodgify UI interface for the
 * Semantic UI Grid.Row.
 *
 * @returns {Object}
 */
export const Component = ({ horizontalAlignContent, ...props }) => (
  <Grid.Row {...props} textAlign={horizontalAlignContent} />
);

Component.displayName = 'GridRow';

Component.defaultProps = {
  horizontalAlignContent: 'left',
};

Component.propTypes = {
  /** Vertically align the content of the column to the bottom, middle or top. */
  horizontalAlignContent: PropTypes.oneOf([
    'left',
    'center',
    'right',
    'justified',
  ]),
};
