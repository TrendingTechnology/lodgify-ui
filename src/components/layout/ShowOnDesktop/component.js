import React from 'react';
import PropTypes from 'prop-types';
import { Responsive } from 'semantic-ui-react';

/**
 * ShowOnDesktop is the Lodgify UI interface for the
 * Semantic UI ShowOnDesktop.
 *
 * To customise how the ShowOnDesktop gets computed, override the @columnCount
 * definition of Semantic UI.
 *
 * @returns {Object}
 */
export const Component = ({ ...props }) => (
  <Responsive {...props} minWidth={599} />
);

Component.displayName = 'ShowOnDesktop';
