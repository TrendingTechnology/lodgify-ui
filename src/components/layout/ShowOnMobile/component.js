import React from 'react';
import PropTypes from 'prop-types';
import { Responsive } from 'semantic-ui-react';

/**
 * ShowOnMobile is the Lodgify UI interface for the
 * Semantic UI ShowOnMobile.
 *
 * To customise how the ShowOnMobile gets computed, override the @columnCount
 * definition of Semantic UI.
 *
 * @returns {Object}
 */
export const Component = ({ ...props }) => (
  <Responsive {...props} maxWidth={599} />
);

Component.displayName = 'ShowOnMobile';
