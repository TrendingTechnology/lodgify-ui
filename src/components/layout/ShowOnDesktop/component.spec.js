import React from 'react';
import { shallow } from 'enzyme';
import { Responsive } from 'semantic-ui-react';

import { expectComponentToHaveProps } from 'lib/expect-helpers';

import { Component as ShowOnDesktop } from './component';

const props = {
  parent: 'ðŸš¸',
};

const getShowOnDesktop = () => shallow(<ShowOnDesktop {...props} />);

describe('<ShowOnDesktop />', () => {
  it('should render a single Semantic UI `Responsive` component', () => {
    const wrapper = getShowOnDesktop();
    const actual = wrapper.is(Responsive);
    expect(actual).toBe(true);
  });

  describe('the `Responsive` component', () => {
    it('it should render the right props', () => {
      const wrapper = getShowOnDesktop()
        .find(Responsive)
        .first();

      expectComponentToHaveProps(wrapper, {
        as: props.parent,
        minWidth: 600,
      });
    });
  });

  it('should have `displayName` `ShowOnDesktop`', () => {
    const actual = ShowOnDesktop.displayName;
    expect(actual).toBe('ShowOnDesktop');
  });
});
