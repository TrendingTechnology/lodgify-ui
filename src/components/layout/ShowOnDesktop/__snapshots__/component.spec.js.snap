// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowOnDesktop /> it should render the right structure 1`] = `
<section
  className="you shall pass mobile-hidden"
>
  div
</section>
`;
